lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -32,21 +126,309 @@ protected function getConfigPrefix()
+		}
+		$url = $this->getConfigValue(array('payment.url-self'));
+		return new \Aimeos\MShop\Common\Item\Helper\Form\Standard($url, 'POST', $list, false, $this->getHTMLform());
+	}
+
+
 	/**
 	 * Returns the value for the given configuration key
 	 *
 	 * @param string $key Configuration key name
 	 * @param mixed $default Default value if no configuration is found
 	 * @return mixed Configuration value
 	 */
-	protected function getValue( $key, $default = null )
+	protected function getValue($key, $default = null)
+	{
+		switch ($key) {
+			case 'type':
+				return 'Stripe';
+			case 'onsite':
+				return true;
+			case 'token':
+				return true;
+			case 'apiKey':
+				return $this->getConfigValue( array( $this->getConfigPrefix() . '.apiKey' ), $default );
+		}
+
+		return parent::getValue($key, $default);
+	}
+
+
+
+
+	/**
+	 * Returns the configuration attribute definitions of the provider to generate a list of available fields and
+	 * rules for the value of each field in the administration interface.
+	 *
+	 * @return array List of attribute definitions implementing \Aimeos\MW\Common\Critera\Attribute\Iface
+	 */
+	public function getConfigBE()
 	{
-		switch( $key )
+		$list = [];
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
+		}
+
+		foreach( $config as $key => $config )
 		{
-			case 'type': return 'Stripe';
-			case 'onsite': return true;
+			$config['code'] = $prefix . '.' . $config['code'];
+			$list[$prefix.'.'.$key] = new \Aimeos\MW\Criteria\Attribute\Standard( $config );
 		}
 
-		return parent::getValue( $key, $default );
+		return $list;
 	}
+
+
+	/**
+	 * Checks the backend configuration attributes for validity.
+	 *
+	 * @param array $attributes Attributes added by the shop owner in the administraton interface
+	 * @return array An array with the attribute keys as key and an error message as values for all attributes that are
+	 * 	known by the provider but aren't valid
+	 */
+	public function checkConfigBE( array $attributes )
+	{
+		$errors = parent::checkConfigBE( $attributes );
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+		$list = [];
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
+		}
+
+		foreach( $config as $key => $cfg )
+		{
+			$cfg['code'] = $prefix . '.' . $cfg['code'];
+			$list[$prefix.'.'.$key] = $cfg;
+		}
+
+		return array_merge( $errors, $this->checkConfig( $list, $attributes ) );
+	}
+
+	/**
+	 * Returns the configuration attribute definitions of the provider to generate a list of available fields and
+	 * rules for the value of each field in the frontend.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $basket Basket object
+	 * @return array List of attribute definitions implementing \Aimeos\MW\Common\Critera\Attribute\Iface
+	 */
+	public function getConfigFE(\Aimeos\MShop\Order\Item\Base\Iface $basket)
+	{
+		return [];
```

Comment:
This method isn't used but should be so it renders the required fields (CC number, expiry, cvv) in the frontend. Then, they can be styled in the frontend and don't need to be passed as part of the HTML string in the HTML client

Reply:
Sorry, a little bit misunderstanding. Is it good now?

Reply:
If the method getConfigFE() will return some fields, they will be render not at the last step, it will be at payment step, something like that: http://joxi.ru/82QeqY5U1eJEg2
But we don't need so. I don't understand that part of Aimeos. How we can use getConfigFE() to render form at the last step only, and prevent it at payment step?

The second one problem is we can't append Stripe element to <input>. So we have to modify templates\checkout\standard\process-body-standard.php or append Stripe element to parent html tag  to `<li class="form-item" >`  (I think it's not good)

Reply:
You are right, `getConfigFE()` is the wrong place and it must be `getPaymentForm()` instead I think.

What's the problem if we tell Stripe to use the `<li class="form-item" >` as container for its input elements? I would expect that it replaces the existing input fields that are inside.

Reply:
Yes it will replace content of `<li class="form-item" >`. Have I to do so and leave at getHtmlForm() Stripe JS only?
I think sometime in the future we can give ability to user to check what Stripe elements he want to show.

Reply:
Cool! Let's do it that way so the surrounding HTML can be changed in the template :-)

Reply:
Then I have to create JS selector for the `li`
`<li class="form-item payment.cardno mandatory">`

I am trying $(".payment\.cardno"); but it does not work with an error at a console: 
>The selector you specified (.payment.cardno) applies to no DOM elements that are currently on the page. Make sure the element exists on the page before calling mount().

It's strange, because if I add an id to that `<li>`, it works fine. 
So I will modify ai-client-html\client\html\templates\checkout\standard\process-body-standard.php to add IDs too.

Reply:
Something with a dot is no valid CSS class. Can you change the output in the HTML template and replace '.' by '-' for the class name?

Reply:
So if we remove fields from GetHtml() and they will be render by $feConfig I want to add `id="payment-<?=$item->getInternalCode();?>"` to `<li>` tag. Is it good?

Reply:
Yes I know it will work. I just fought it's not good to use '-' instead of dot because at all payment providers at Aimeos you are using dot as standard delimeter.

Reply:
You must not change the keys in your Stripe class but mangle the output in the HTML client template:
https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/checkout/standard/process-body-standard.php#L129

Reply:
Simply add a `str_replace( '.', '-', $key );` there:
https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/checkout/standard/process-body-standard.php#L129

Reply:
Also as you can see at my JS I'm calling an AimeosPurchaseHandler object. It's for the future, if we will create card details form at the chosing provider step:  http://joxi.ru/xAeY510fYQNk0A

Reply:
We try to avoid IDs as they need to be unique over the page and CSS classes don't have to. This is especially a concern when things can be dynamically extended by extensions.

Can you use `document.getElementsByClassName("...");` instead of jQuery so we don't need it?

I would suggest to remove the functionality of AimeosPurchaseHandler for the moment because it doesn't seem to be generic enough to be included into the Aimeos admin.js. You can create another PR where we can discuss how to integrate Stripe into the payment section of the checkout.

Reply:
Except button's ID. We have added id="payment-button" to Pay Now button already

Reply:
Please note one more thing. We have added `<script src="https://js.stripe.com/v3/"></script>`
just to checkout page, by getHtmlForm() method of form helper. But according to Stripe documentation it's recommended to place this including JS to the each one page to prevent fraud. 
I don't know how to do here. 
I still think we need some "hook" to place any necessary code to the footer or header.

Reply:
I would ignore that recommendation for the moment because it doesn't make sense to have that file at each page. It's possible to place that in the HTML head if you add it to the process-header-standard template. But you would have to create a decorator for this that adds that part to the view and the template has to render that information afterwards.

Reply:
Yes, I don't insist on it, just notice. Or maybe we can explain it as description of provider to users who will use it.